### DISCUSSION TITLE
Distinguishing product variation types in REST API responses

### DISCUSSION THREAD

**Discussion by mattsherman** [woocommerce/woocommerce#47420] [General]
In a recent [developer blog comment](https://developer.woocommerce.com/2023/11/17/getting-to-know-woo-extensibility-in-the-new-product-editor/#comment-20676), a third party developer expressed the need to distinguish between product variation types.

Historically, we haven't included the `type` property for product variation REST API responses.

After looking into this, I found that it is "easy" to add such support to the REST API. But, I don't know if there are reasons why this was never done and if it has hidden issues.

PR that implements support:

- https://github.com/woocommerce/woocommerce/pull/47377

I'd love to get some feedback from both core devs (@woocommerce/proton and @woocommerce/mothra in particular) and third party devs on this PR (and the higher-level topic of distinguishing variation types), since while it is a very small code change, it could have big implications!

**Comment by nathanss** [comment]
I'm in favor of this. I feel like not having the `type` property can create a lot of edge cases and inconsistencies. I'm currently migrating an extension and I'm looking for a way to match the correct template to the product, and having the `type` specified in `product_data` ([like here](https://github.com/woocommerce/woocommerce/blob/e65481aa274b6faf3e03140dc42ab5a599f11867/plugins/woocommerce/src/Admin/Features/ProductBlockEditor/Init.php#L249)) would help with this.

Also, not having the property might harm reusability of blocks, for example, if a block checks for the `type` property, it currently cannot be reused in variations, since it will be `undefined`.

**Comment by barryhughes** [comment]
If we can do this in a back-compat friendly manner (based on the description here and a very quick look at the linked PR, it seems we can) I think this will be a nice addition.

**Comment by mattsherman** [comment]
Closing this discussion as the functionality was merged in the above mentioned PR.
